mathics/builtin/quantities.py

```diff
@@ -185,88 +175,63 @@ class QuantityMagnitude(Builtin):
 
     >> QuantityMagnitude[Quantity[{10,20}, "Meters"]]
      = {10, 20}
-
-    #> QuantityMagnitude[Quantity[1, "meter"], "centimeter"]
-     = 100
-
-    #> QuantityMagnitude[Quantity[{3,1}, "meter"], "centimeter"]
-     = {300, 100}
-
-    #> QuantityMagnitude[Quantity[{300,100}, "centimeter"], "meter"]
-     = {3, 1}
-
-    #> QuantityMagnitude[Quantity[{3, 1}, "meter"], "inch"]
-     = {118.11, 39.3701}
-
-    #> QuantityMagnitude[Quantity[{3, 1}, "meter"], Quantity[3, "centimeter"]]
-     = {300, 100}
-
-    #> QuantityMagnitude[Quantity[3,"mater"]]
-     : Unable to interpret unit specification mater.
-     = QuantityMagnitude[Quantity[3,mater]]
     """
 
     summary_text = "get magnitude associated with a quantity."
 
-    def eval(self, expr, evaluation: Evaluation):
-        "QuantityMagnitude[expr_]"
+    def eval_List1(self, expr, evaluation: Evaluation):
+        "QuantityMagnitude[expr_List]"
+        return ListExpression(
+            *(
+                Expression(Symbol(self.get_name()), e).evaluate(evaluation)
+                for e in expr.elements
+            )
+        )
+
+    def eval_List2(self, expr, unit, evaluation: Evaluation):
+        "QuantityMagnitude[expr_List, unit_]"
+        return ListExpression(
+            *(
+                Expression(Symbol(self.get_name()), e, unit).evaluate(evaluation)
+                for e in expr.elements
+            )
+        )
 
-        def get_magnitude(elements):
```

Comment:
Here I tried to disentangle this by adding specific methods for each different form.